xray_core/src/buffer.rs

```diff
@@ -242,6 +294,17 @@ impl Buffer {
             text
         })
     }
+
+    pub fn as_register_stream(&self) -> RegisterStream<MPMCFutReceiver<Version>> {
+        RegisterStream(self.change_events_queue.receiver.clone())
+    }
+
+    fn broadcast_change(&self) {
+        match self.change_events_queue.sender.try_send(self.lamport_clock) {
+            Err(TrySendError::Full(_)) => panic!("Tried to broadcast a change on a full queue"),
```

Comment:
I guess this situation shouldn't end up happening, but I wonder what we should do if it does. I worry about this biting us later in rare circumstances.

It seems like the docs suggest some kind of side buffer that we manage ourselves. Like could we try to send again after 100ms or something? Structuring that to get added to the event loop could be a bit weird though in this layer, since we don't have access to a specific loop implementation. Maybe we should pass one in when the buffer is constructed.